include/seqan3/std/ranges

```diff
@@ -280,4 +295,646 @@ constexpr bool enable_view<type> = static_cast<bool>(seqan3::detail::enable_view
+ * \param[in]        urange The range being processed. [parameter is omitted in pipe notation]
+ * \param[in,out] predicate The predicate.
+ * \returns A range of those elements in the underlying range that satisfy the predicate.
+ * \ingroup ranges
+ *
+ * ### View properties
+ *
+ * | range concepts and reference_t  | `urng_t` (underlying range type)      | `rrng_t` (returned range type)                     |
+ * |---------------------------------|:-------------------------------------:|:--------------------------------------------------:|
+ * | std::ranges::InputRange         | *required*                            | *preserved*                                        |
+ * | std::ranges::ForwardRange       |                                       | *preserved*                                        |
+ * | std::ranges::BidirectionalRange |                                       | *preserved*                                        |
+ * | std::ranges::RandomAccessRange  |                                       | *lost*                                             |
+ * | std::ranges::ContiguousRange    |                                       | *lost*                                             |
+ * |                                 |                                       |                                                    |
+ * | std::ranges::ViewableRange      | *required*                            | *guaranteed*                                       |
+ * | std::ranges::View               |                                       | *guaranteed*                                       |
+ * | std::ranges::SizedRange         |                                       | *lost*                                             |
+ * | std::ranges::CommonRange        |                                       | *lost*                                             |
+ * | std::ranges::OutputRange        |                                       | *preserved*                                        |
+ * | seqan3::const_iterable_concept  |                                       | *lost*                                             |
+ * |                                 |                                       |                                                    |
+ * | seqan3::reference_t             |                                       | seqan3::reference_t<urng_t>                        |
+ *
+ * See the \link view view submodule documentation \endlink for detailed descriptions of the view properties.
+ *
+ * ### Example
+ *
+ * ```cpp
+ * dna5_vector s{"ACTNTGATAN"_dna5};
+ * auto v1 = s | std::view::filter([](dna5 const l) { return (l != 'N'_dna5); }); // == "ACTTGATA"_dna5
+ * ```
+ * \hideinitializer
+ */
+inline constexpr auto filter = ::ranges::view::filter;
+
+} // namespace std::ranges::view
+
+namespace std::ranges
+{
+
+/*!\brief               A view over an input stream.
```

Comment:
```suggestion
/*!\brief               Makes the underlying range behave like an input range.
```
? 
The brief right now does not really tell you anything. You can also use view::all to make a view over an input stream.

Reply:
> The brief right now does not really tell you anything. 

It tells you all you need :laughing: 

> You can also use subrange to make a view over an input stream.

This is a hack though. You can see that in our I/O, that we always need to do the weird check at the end, whether its actually at end, and increment once more et cetera.

Once these things are documented on en.cppreference.com we will remove our documentation anyway and link there.

Reply:
what does it do to no-input ranges?

Reply:
I understand neither of these questions :see_no_evil:

Reply:
Ignore the first.

2. what happens: `istream_view(std::vector<int>)`?

Reply:
You can't. The documentation states that the view is source-only and that it has exactly one parameter:
```
 * \param[in] istr      The object of type std::istream.
```

Reply:
ah ok.